frontend/templates/product/sections/ProductSection/CompatibleDevices.tsx

```diff
@@ -0,0 +1,55 @@
+import { Product } from '@models/product';
+import NextLink from 'next/link';
+import { CompatibleDevice } from '@components/common';
+import { chakra, Link } from '@chakra-ui/react';
+
+export type CompatibleDevicesProps = {
+   product: Product;
+};
+
+export function CompatibleDevices({ product }: CompatibleDevicesProps) {
+   const compatibilityDrawerModelsTruncate = 4;
+   const compatibilityDrawerDeviceTruncate = 3;
+   const compatibilityDrawerIncomplete =
+      product.compatibility &&
+      (product.compatibility.devices.length >
+         compatibilityDrawerDeviceTruncate ||
+         product.compatibility.devices.some(
+            (currentValue) =>
+               currentValue.variants.length > compatibilityDrawerModelsTruncate
+         ));
```

Comment:
This code might be improved a little bit

```suggestion
const MAX_VISIBLE_MODELS_PER_DEVICE = 4;

const MAX_VISIBLE_DEVICES = 3;

export function CompatibleDevices({ product }: CompatibleDevicesProps) {
  const deviceCount = product.compatibility?.devices.length ?? 0;
  
  const hasMoreDevicesToShow = 
    deviceCount > MAX_VISIBLE_DEVICES ||
    product.compatibility?.devices.some(
       (device) => device.variants.length > MAX_VISIBLE_MODELS_PER_DEVICE
    );
```